### PR TITLE
Fix #4841: Load the compiler toolbox reflectively

### LINKED ISSUE #4841: Don’t expose compiler APIs in TASTY macros

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by smarter** [comment]
Can you rebase this PR on top of https://github.com/lampepfl/dotty/pull/4833 to make sure that it works as expected with the stricter checks in that PR (or even better review and merge it :)).

File: library/src/scala/quoted/Toolbox.scala [hunk 1]
```diff
@@ -2,8 +2,56 @@ package scala.quoted
...
+    val cl = getClass.getClassLoader
+    try {
+      val toolboxImplCls = cl.loadClass("dotty.tools.dotc.quoted.ToolboxImpl")
+      val makeMeth = toolboxImplCls.getMethod("make", classOf[Settings])
+      makeMeth.invoke(null, settings).asInstanceOf[Toolbox]
+    }
+    catch {
+      case ex: ClassNotFoundException =>
+        throw new ToolboxNotFoundException(
+          s"""Could not load the Toolbox class `${ex.getMessage}` in the classpath. Make sure that the compiler is on the classpath.""",
```
Comment (smarter) [inline library/src/scala/quoted/Toolbox.scala]: "in the classpath" -> "from the JVM classpath"
"on the classpath" -> "on the JVM classpath"
(To distinguish it from the compiler classpath)

Comment (nicolasstucki) [reply library/src/scala/quoted/Toolbox.scala]: Changed

**Comment by nicolasstucki** [comment]
I got some unpickling issues when I moved the tests back.

**Comment by allanrenucci** [comment]
> I got some unpickling issues when I moved the tests back.

Can you investigate? It would be nice that the tests enforce that the compiler is not needed on the classpath

**Pull request merged**